test/unit-test/ota_utest.c

```diff
@@ -1920,3 +1931,61 @@ void test_OTA_HTTP_strerror( void )
     str = OTA_HTTP_strerror( status );
     TEST_ASSERT_EQUAL_STRING( "InvalidErrorCode", str );
 }
+
+/**
+ * @brief Test that setDataInterface sets the data interface when given valid
+ *        inputs.
+ */
+void test_OTA_setDataInterfaceValidInput( void )
+{
+    OtaDataInterface_t dataInterface = { NULL, NULL, NULL, NULL };
+    uint8_t pProtocol[ OTA_PROTOCOL_BUFFER_SIZE ] = { 0 };
+
+    memcpy( pProtocol, "MQTT", sizeof( "MQTT" ) );
+    TEST_ASSERT_EQUAL( OtaErrNone, setDataInterface( &dataInterface, pProtocol ) );
+    TEST_ASSERT_EQUAL( initFileTransfer_Mqtt, dataInterface.initFileTransfer );
+    TEST_ASSERT_EQUAL( requestFileBlock_Mqtt, dataInterface.requestFileBlock );
+    TEST_ASSERT_EQUAL( decodeFileBlock_Mqtt, dataInterface.decodeFileBlock );
+    TEST_ASSERT_EQUAL( cleanupData_Mqtt, dataInterface.cleanup );
+
+    memcpy( pProtocol, "HTTP", sizeof( "HTTP" ) );
+    memset( &dataInterface, 0, sizeof( dataInterface ) );
+    TEST_ASSERT_EQUAL( OtaErrNone, setDataInterface( &dataInterface, pProtocol ) );
+    TEST_ASSERT_EQUAL( initFileTransfer_Http, dataInterface.initFileTransfer );
+    TEST_ASSERT_EQUAL( requestDataBlock_Http, dataInterface.requestFileBlock );
+    TEST_ASSERT_EQUAL( decodeFileBlock_Http, dataInterface.decodeFileBlock );
+    TEST_ASSERT_EQUAL( cleanupData_Http, dataInterface.cleanup );
+
+    memcpy( pProtocol, "HTTPMQTT", sizeof( "HTTPMQTT" ) );
+    memset( &dataInterface, 0, sizeof( dataInterface ) );
+    TEST_ASSERT_EQUAL( OtaErrNone, setDataInterface( &dataInterface, pProtocol ) );
+    TEST_ASSERT_NOT_EQUAL( NULL, dataInterface.initFileTransfer );
+    TEST_ASSERT_NOT_EQUAL( NULL, dataInterface.requestFileBlock );
+    TEST_ASSERT_NOT_EQUAL( NULL, dataInterface.decodeFileBlock );
+    TEST_ASSERT_NOT_EQUAL( NULL, dataInterface.cleanup );
+
+    memcpy( pProtocol, "MQTTHTTP", sizeof( "MQTTHTTP" ) );
+    memset( &dataInterface, 0, sizeof( dataInterface ) );
+    TEST_ASSERT_EQUAL( OtaErrNone, setDataInterface( &dataInterface, pProtocol ) );
+    TEST_ASSERT_NOT_EQUAL( NULL, dataInterface.initFileTransfer );
+    TEST_ASSERT_NOT_EQUAL( NULL, dataInterface.requestFileBlock );
+    TEST_ASSERT_NOT_EQUAL( NULL, dataInterface.decodeFileBlock );
+    TEST_ASSERT_NOT_EQUAL( NULL, dataInterface.cleanup );
+}
+
+/**
+ * @brief Test that setDataInterface returns an error and does not set the data
+ * interface when provided with an invalid input from a job document.
+ */
+void test_OTA_setDataInterfaceInvalidInput( void )
+{
+    OtaDataInterface_t dataInterface = { NULL, NULL, NULL, NULL };
+    uint8_t pProtocol[ 20 ] = { 0 };
+
+    memcpy( pProtocol, "invalid_protocol", sizeof( "invalid_protocol" ) );
```

Comment:
I'd add test cases with some garbage/unexpected values for `pProtocol` which contain `HTTP` and/or `MQTT` in here.  Those will fail with the `strstr()` implementation above.

Reply:
Done in this commit
https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/152/commits/b76cae926d1feacca12a065b071421ec1098a154